packages/nimble-components/src/select/specs/filterable-select-hld.md

```diff
@@ -20,29 +20,58 @@ Our clients have a need of a filterable dropdown component that does not allow a
 
 ```ts
 export class Select() {
     ...
     @attr({ attribute: 'filter-mode' })
     public filterMode = FilterMode.none;
+
+    /**
+     * Displays a message at the bottom of the dropdown to indicate more options
+     * are currently being loaded.
+     */
+    @attr()
+    public loading = false;
     ...
 }
 
 // Provide enum for filterMode to allow for future modes
 export const FilterMode  = {
     none: undefined;
     standard: 'standard';
+    dynamic: 'dynamic';
 } as const;
 ```
 
 -   The `standard` filterMode will result in case-insensitive, diacritic-insensitive filtering.
--   `filterMode` will default to `none` so as not to affect existing clients.
+-   The `dynamic` filterMode will display a search input, but it won't automatically filter the options as the user types. Instead, clients can use the `filter-input` event described in the ['Dynamic Options'](#dynamic-options) section below to implement their own filtering logic.
 -   The `none` filter mode results in no search input being shown in the dropdown.
+-   `filterMode` will default to `none` so as not to affect existing clients.
 
 _Note: The `filterMode` isn't meant to mirror the `Combobox` `autocomplete` API, as they do serve slightly different purposes: The `autocomplete` for the `Combobox` ultimately helps set the actual value of the `Combobox` as the user types, and isn't necessarily performing any filtering (e.g. the `inline` mode). One possible concern, however, is that we are presenting an API that will allow different types of filter behaviors (i.e. case sensitive) that the `Combobox` does not support. Additionally, I am proposing diacritic insensitive filtering, which the `Combobox` also does not currently support, but I feel this is quite likely a better default experience._
 
+#### Dynamic options
+
+The `filterMode: dynamic` option allows clients to dynamically supply list options to the `select` control in response to user search text, by subscribing to the `filter-input` event.
+
+```ts
+interface SelectFilterInputEventDetail {
+    filterText: string;
+}
+```
+
+Then, when handling the `filter-input` event, clients should perform the desired filtering using the provided filter text. Additionally, they should set the `loading` attribute to `true` while the options are being loaded into the DOM, and then set it back to `false` when the loading process is complete. A `filter-input` event will also be emitted with its `filterText` set to empty string upon closure of the dropdown if both the `filter-mode` was not set to `none` _and_ the filter input contains non-empty text.
+
+_IMPORTANT_: When using the `dynamic` filter mode, clients are responsible for making sure the options that are placed in the DOM meet one of the following criteria:
+
+-   The option is some reasonable default option when there is no filter text provided _OR_
+-   The option display text matches the provided filter in an understandable way _OR_
+-   The option is a placeholder option
```

Comment:
It's clear to me the implications of this requirement. Conceptually changing out options is independent of the filter-mode. Programmatically a user could do it at any time. It shouldn't be unique to filter mode. And it's a declarative state, whatever is slotted in has a meaning and well-defined behavior.

The only thing I could think of is that user's should be aware that they need to keep whatever item is currently selected in the options list as selected, even if it doesn't match their custom filter algorithm so it continues to show as the current display text. Otherwise they will accidentally trigger a value change (but not a value change event).

And if they have something configured as a placeholder and they want it to keep showing as a placeholder, they need to makee sure it does not disappear in the list (and may need to make sure it stays first in the list / they don't mark multiple things `disabled hidden`).

Reply:
Updated.